heartbeat/simple_detector_test.go

```diff
@@ -49,3 +49,13 @@ func (suite *SimpleDetectorSuite) TestDetectPropogatesValues() {
 	suite.Assert().Equal(expired, []string{"foo", "bar"})
 	suite.Assert().Equal(err.Error(), "baz")
 }
+
+func (suite *SimpleDetectorSuite) TestDetectPurgesData() {
+	strategy := &TestStrategy{}
+	strategy.On("Purge", "foo", "id1", suite.Pool).Return(nil).Once()
+	strategy.On("Purge", "foo", "id2", suite.Pool).Return(errors.New("baz")).Once()
+
+	d := heartbeat.NewDetector("foo", suite.Pool, strategy)
+	suite.Assert().Nil(d.Purge("id1"))
```

Comment:
I've found that splitting tests into:

1. Setup
2. Action/mutation
3. Assertions

By new-line is helpful for clarity.